flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/exceptions.scala

```diff
@@ -165,3 +165,31 @@ case class AmbiguousTableSourceConverterException(
 
   def this(tableType: String) = this(tableType, null)
 }
+
+/**
+  * Exception for operation on a nonexistent external catalog
+  *
+  * @param catalogName external catalog name
+  * @param cause
+  */
+case class ExternalCatalogNotExistException(
```

Comment:
Maybe we should introduce some superclasses, because there are many exceptions that users might want to catch at once.